frontend/packages/console-dynamic-plugin-sdk/src/extensions/notification-alert.ts

```diff
@@ -0,0 +1,18 @@
+import { Extension } from '@console/plugin-sdk/src/typings/base';
+import { ExtensionDeclaration, CodeRef } from '../types';
+
+export type AlertAction = ExtensionDeclaration<
+  'console.alert-action',
+  {
+    /* Alert name as defined by `alert.rule.name` property */
+    alert: string;
+    /* Action text */
+    text: string;
+    /* Action is a link or a function that performs side effects(Ex: Opens a modal). */
+    isLink: boolean;
+    /* function that returns the path or performs side effects */
+    actionProperty: CodeRef<(alert) => string | void>;
```

Comment:
Why don't you do `history.push` for links and keep just one path resolver ?

Reply:
`no-implicit-globals` would not allow the history object to be directly accessed in the utils folder. We could obviously disable it. @christianvogt wdyt?

Reply:
Can you explain more about the use case here ? I am failing to interpret the above statement wrt `utils` folder.

Reply:
In general links should be rendered as a link and not a button. However I'm not opposed to consistency here where the action itself simply handles navigation or whatever else it wants to do instead. It would simplify the API.

You can import the `history` from `'@console/internal/components/utils';`